pkg/data/management/fleet_roles.go

```diff
@@ -0,0 +1,343 @@
+package management
+
+import (
+	"reflect"
+
+	"github.com/rancher/rancher/pkg/types/config"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ensureFleetRoles(management *config.ManagementContext) error {
+	fleetWorkspaceAdminRole := v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "fleetworkspace-admin",
+		},
+		Rules: []v1.PolicyRule{
+			{
+				APIGroups: []string{
+					"fleet.cattle.io",
+				},
+				Resources: []string{
+					"gitrepos",
```

Comment:
For admin this can just be `*` for all resources in the `fleet.cattle.io` group.